tests/rest/client/v1/test_login.py

```diff
@@ -47,8 +47,9 @@
     HAS_JWT = False
 
 
-# public_base_url used in some tests
-BASE_URL = "https://synapse/"
+# synapse server name: used to populate public_base_url in some tests
+SYNAPSE_SERVER_PUBLIC_HOSTNAME = "synapse"
+BASE_URL = "http://%s/" % (SYNAPSE_SERVER_PUBLIC_HOSTNAME,)
```

Comment:
swapping to `http` here, because `FakeChannel.isSecure()` returns `False`, so synapse will see the requested uri as `http://synapse/...`.  Configuring that as the `public_baseurl` avoids the redirect.

Reply:
fair enough. I've updated some comments.